yt_dlp/extractor/archiveorg.py

```diff
@@ -588,48 +749,79 @@ def _extract_metadata(self, video_id, webpage):
             video_details.get('title')
             or YoutubeBaseInfoExtractor._get_text(microformats, 'title')
             or YoutubeBaseInfoExtractor._get_text(initial_data_video, 'title')
+            or traverse_obj(player_config, ('args', 'title'))
             or self._extract_webpage_title(webpage)
             or search_meta(['og:title', 'twitter:title', 'title']))
 
+        def id_from_url(url, type_):
+            return self._search_regex(
+                rf'(?:{type_})/([^/#&?]+)', url or '', f'{type_} id', default=None)
+
+        # XXX: would the get_elements_by_... functions be better suited here?
+        # Uploader ID and URL
+        _CHANNEL_URL_HREF_RE = r'href="[^"]*(?P<url>https?://www\.youtube\.com/(?:user|channel)/[^"]+)"'
+        upch_url = self._search_regex(
+            [fr'<(?:link\s*itemprop=\"url\"|a\s*id=\"watch-username\").*?\b{_CHANNEL_URL_HREF_RE}>',  # @fd05024
+             fr'<div\s*id=\"(?:watch-channel-stats|watch-headline-user-info)\"[^>]*>\s*<a[^>]*\b{_CHANNEL_URL_HREF_RE}'],  # ~ May 2009, ~June 2012
+            webpage, 'uploader or channel url', default=None)
+
+        owner_profile_url = url_or_none(microformats.get('ownerProfileUrl'))  # @a6211d2
+
+        # Uploader refers to the /user/ id ONLY
+        uploader_id = (
+            id_from_url(owner_profile_url, 'user')
+            or id_from_url(upch_url, 'user')
+            or ytcfg.get('VIDEO_USERNAME'))
+        uploader_url = f'https://www.youtube.com/user/{uploader_id}' if uploader_id else None
+
+        # TODO: sort out what uploader means
+        uploader = (
+            self._search_regex(
+                [r'<a\s*id=\"watch-username\".*\">\s*<strong[^>]?>([^<]+)</strong>',
```

Comment:
u sure it should be greedy?
```suggestion
                [r'<a\s*id="watch-username".*?">\s*<strong[^>]?>([^<]+)</strong>',
```